ui/StatusQ/src/StatusQ/Components/StatusPageIndicator.qml

```diff
@@ -0,0 +1,138 @@
+import QtQuick 2.14
+import QtQuick.Controls 2.14
+
+import StatusQ.Core.Theme 0.1
+import StatusQ.Controls 0.1
+
+/*!
+   \qmltype StatusPageIndicator
+   \inherits Item
+   \inqmlmodule StatusQ.Components
+   \since StatusQ.Components 0.1
+   \brief Displays page indicator, clicking on a certain page number, it becomes selected and `currentIndex` is set accordingly.
+
+   Example:
+
+   \qml
+    StatusPageIndicator {
+        totalPages: 10
+        currentIndex: 5
+    }
+   \endqml
+
+   \image status_page_indicator.png
+
+   For a list of components available see StatusQ.
+*/
+
+Item {
```

Comment:
This could shave off a lot of logic by just inheriting https://doc.qt.io/qt-5/qml-qtquick-controls2-pageindicator.html and customizing its `contentItem/delegate` (see https://doc.qt.io/qt-5/qtquickcontrols2-customize.html#customizing-pageindicator)

Reply:
This is what we need:
https://www.figma.com/file/FkFClTCYKf83RJWoifWgoX/Wallet-v2?node-id=12272-268295&t=npET52VwEBU9rkFd-0

So for 20 pages at the start it should be:
- Selected 1st page: `| 1 | 2 | 3 | ... | 20 |`
- Selected 10th page: `| 1 | ... | 9 | 10 | 11 | ... | 20 |`

Reply:
Yeah, that's the `delegate` (single item), which in your case would be either a number or "...". You'd probably have to replace the `contentItem` too

Reply:
But that delegat would be instantiated e.g. 20 times for 20 pages, and that's not what we need. That's why I think that logic won't help me.

Reply:
Right, you'd still have to somehow compute the `model` yourself, an example:

```qml
    contentItem: Row {
        spacing: control.spacing

        Repeater {
            model: control.count
            delegate: control.delegate
        }
    }
```

I'll leave it up to you whether you still want to pursuit this or not :)